src/main/java/com/google/step/servlets/HomePageServlet.java

```diff
@@ -0,0 +1,111 @@
+  private final TagManager tagManager;
+  private final FollowManager followManager;
+
+  public HomePageServlet(
+      DealManager dealManager,
+      UserManager userManager,
+      RestaurantManager restaurantManager,
+      VoteManager voteManager,
+      DealTagManager dealTagManager,
+      TagManager tagManager,
+      FollowManager followManager) {
+    this.dealManager = dealManager;
+    this.userManager = userManager;
+    this.voteManager = voteManager;
+    this.restaurantManager = restaurantManager;
+    this.dealTagManager = dealTagManager;
+    this.tagManager = tagManager;
+    this.followManager = followManager;
+  }
+
+  public HomePageServlet() {
+    dealManager = new DealManagerDatastore();
+    userManager = new UserManagerDatastore();
+    voteManager = new VoteManagerDatastore();
+    restaurantManager = new RestaurantManagerDatastore();
+    tagManager = new TagManagerDatastore();
+    dealTagManager = new DealTagManagerDatastore();
+    followManager = new FollowManagerDatastore();
+  }
+
+  /** Gets the deals for the home page */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long userId = 1; // TODO get authenticated user id
+    List<Deal> trendingDeals = dealManager.getTrendingDeals();
+    List<Deal> dealsByUsersFollowed =
+        dealManager.getDealsPublishedByFollowedUsers(followManager.getFollowedUserIds(userId));
+    List<Deal> dealsByRestaurantsFollowed =
+        dealManager.getDealsPublishedByFollowedRestaurants(
+            followManager.getFollowedRestaurantIds(userId));
+    List<Deal> dealsByTagsFollowed =
+        dealManager.getDealsPublishedByFollowedTags(followManager.getFollowedTagIds(userId));
+    List<List<Deal>> homePageDeals =
+        new ArrayList<>(
+            Arrays.asList(
+                trendingDeals,
+                dealsByUsersFollowed,
+                dealsByRestaurantsFollowed,
+                dealsByTagsFollowed));
+    response.setContentType("application/json;");
+    response.getWriter().println(JsonFormatter.getHomePageJson(getHomePageDealMaps(homePageDeals)));
+  }
+
+  /** Creates a list of list of deal maps for the home page */
+  private List<List<Map<String, Object>>> getHomePageDealMaps(List<List<Deal>> homePageDeals) {
+    List<List<Map<String, Object>>> homePageDealsMapList = new ArrayList<>();
+    for (List<Deal> dealList : homePageDeals) {
+      List<Map<String, Object>> homePageSectionDealMaps = new ArrayList<>();
+      for (Deal deal : dealList) {
+        User user = userManager.readUser(deal.posterId);
+        Restaurant restaurant = restaurantManager.readRestaurant(deal.restaurantId);
+        List<Tag> tags = tagManager.readTags(dealTagManager.getTagIdsOfDeal(deal.id));
+        int votes = voteManager.getVotes(deal.id);
+        Map<String, Object> homePageDealMap =
+            JsonFormatter.getBriefHomePageDealMap(deal, user, restaurant, tags, votes);
```

Comment:
don't call getMap from the JsonFormatter. The getMap methods should be private. Get the JSON string directly instead.